lib/msal-core/src/ServerRequestParameters.ts

```diff
@@ -355,6 +351,29 @@ export class ServerRequestParameters {
         return paramsString;
     }
 
+    /**
+     * @hidden
+     *
+     * Validate  scopes/state/correlationId set in the request by the user
+     * @param request
+     */
+    private validateUserSetServerParams(scopes: Array<string>, state: string, correlationId: string, clientId: string) {
```

Comment:
If the function is called validate, I wouldn't expect it to create a new object (see validateScopes). I would create the objects and pass it in to validate, or rename the function. You can call it createValidServerParams or something and then set the field variables (this.state, this.correlationid, etc.) directly there instead of returning an object from the function.

Reply:
it should probably be setValidServerParams instead of create"".

Reply:
I agree. I missed that these are this.* params and can be set in a different function. Changes made.